src/trie/merkle_tree.rs

```diff
@@ -112,6 +118,39 @@ fn test_shared_path_encode_decode() {
     let decoded = Path::decode(&mut &encoded[..]).unwrap();
     assert_eq!(path, decoded);
 }
+/// A node used in proof generated by the trie.
+///
+/// See pathfinders merkle-tree crate for more information.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ProofNode {
+    Binary {
+        left: Felt252Wrapper,
+        right: Felt252Wrapper,
+    },
+    Edge {
+        child: Felt252Wrapper,
+        path: Path,
+    },
+}
+
+impl ProofNode {
+    pub fn hash<H: HasherT>(&self) -> Felt252Wrapper {
+        match self {
+            ProofNode::Binary { left, right } => Felt252Wrapper(H::hash_elements(left.0, right.0)),
+            ProofNode::Edge { child, path } => {
+                let mut bytes = [0u8; 32];
+                bytes.view_bits_mut::<Msb0>()[256 - path.0.len()..].copy_from_bitslice(&path.0);
+                let path_hash = Felt252Wrapper::try_from(&bytes).unwrap();
```

Comment:
can you add explain why it's safe to unwrap

Reply:
Done

Reply:
Felt is slightly less than 32 bytes it's 252 bits so 32 bytes isn't always safe

Reply:
AFAIU it's safe because the path is 251 bits which is < than our prime

Reply:
Yes sorry I mixed up multiple things. Corrected.